File: src/WellTool.Jwt/hutool-jwt/src/main/java/cn/hutool/jwt/signers/NoneJWTSigner.java
package cn.hutool.jwt.signers;

import cn.hutool.core.util.StrUtil;

/**
 * 无需签名的JWT签名器
 *
 * @author looly
 * @since 5.7.0
 */
public class NoneJWTSigner implements JWTSigner {

	/**
	 * 定义一个常量ID_NONE，表示没有ID的情况
	 */
	public static final String ID_NONE = "none";

	/**
	 * 创建一个NoneJWTSigner实例，用于处理没有签名的JWT
	 */
	public static NoneJWTSigner NONE = new NoneJWTSigner();

	/**
	 * 判断给定的算法是否为无签名的算法
	 *
	 * @param alg 算法
	 * @return 如果是无签名的算法，则返回true；否则返回false
	 * @since 5.8.42
	 */
	public static boolean isNone(final String alg) {
		return StrUtil.isBlank( alg) || StrUtil.equalsIgnoreCase(alg, ID_NONE);
	}

	@Override
	public String sign(String headerBase64, String payloadBase64) {
		return StrUtil.EMPTY;
	}

	@Override
	public boolean verify(String headerBase64, String payloadBase64, String signBase64) {
		return StrUtil.isEmpty(signBase64);
	}

	@Override
	public String getAlgorithm() {
		return ID_NONE;
	}
}

File: src/WellTool.Jwt/Signers/NoneJwtSigner.cs
namespace WellTool.JWT.Signers;

/// <summary>
/// 无签名JWT签名器
/// </summary>
public class NoneJwtSigner : IJwtSigner
{
    /// <summary>
    /// 无签名实例
    /// </summary>
    public static readonly NoneJwtSigner None = new();

    /// <summary>
    /// 私有构造函数
    /// </summary>
    private NoneJwtSigner()
    {
    }

    /// <summary>
    /// 签名
    /// </summary>
    /// <param name="headerBase64">Base64编码的头部</param>
    /// <param name="payloadBase64">Base64编码的载荷</param>
    /// <returns>空字符串</returns>
    public string Sign(string headerBase64, string payloadBase64)
    {
        return string.Empty;
    }

    /// <summary>
    /// 验证签名
    /// </summary>
    /// <param name="headerBase64">Base64编码的头部</param>
    /// <param name="payloadBase64">Base64编码的载荷</param>
    /// <param name="signatureBase64">Base64编码的签名</param>
    /// <returns>是否验证通过</returns>
    public bool Verify(string headerBase64, string payloadBase64, string signatureBase64)
    {
        return string.IsNullOrEmpty(signatureBase64);
    }

    /// <summary>
    /// 获取算法名称
    /// </summary>
    /// <returns>算法名称</returns>
    public string GetAlgorithm()
    {
        return "none";
    }

    /// <summary>
    /// 获取算法ID
    /// </summary>
    /// <returns>算法ID</returns>
    public string GetAlgorithmId()
    {
        return "none";
    }

    /// <summary>
    /// 判断算法是否为none
    /// </summary>
    /// <param name="algorithm">算法名称</param>
    /// <returns>是否为none</returns>
    public static bool IsNone(string algorithm)
    {
        return "none".Equals(algorithm, StringComparison.OrdinalIgnoreCase);
    }
}
